### DISCUSSION TITLE
Sidebar keeps showing on startup

### DISCUSSION THREAD

**Discussion by s3cret000** [aclap-dev/video-downloadhelper#865] [Help and Q&A]
### Your question

Hello,

This is on Firefox.

Whenever I start Firefox, the sidebar for VDH is also there. This includes when I disable the extension.
Any way to keep it in pop-up mode?

Thanks

### Addon & CoApp version and details

version: 9.0.2.3
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 5
}

**Comment by GrampaWildWilly** [comment]
You should search this forum with "sidebar" as your search key.  Make sure your Filter is set to All, not some other value.

**Reply by s3cret000** [reply]
I did see the other thread (which is how it led me to GitHub in the first place), but didn't see the hidden replies in the middle.

I guess it is a Firefox issue, so would rather keep this open until it's addressed and fixed.

**Reply by GrampaWildWilly** [reply]
Not entirely sure about that.  If you're referring to #840, read it again carefully.  Yes, Paul cites an internal Mozilla bug discussion.  But the real problem there was solved by the user actually using the product properly.  There are not a lot of people reporting this issue so I have to think the odds are that you are doing something wrong.  You should make sure you are following the technique I show with screenshots in #637.  But surely you already found that one by using "sidebar" as your search key.

As for the "Show x previous replies" thing, I hate those, too.  But that's the way GitHub does things so you just have to remember those might be there & always click them.

**Reply by s3cret000** [reply]
Okay, help me understand how to use the product here.

VDH is installed.
Anytime I open Firefox, the VDH sidebar is open. Including when I disable the add-on itself. I don't want it there when I open Firefox.
I can click the 'Show in popup' button or the 'Close sidebar' button, or the VDH dropdown and click the 'Close sidebar' option to make it disappear, but I don't want to have to do that every single time I open Firefox.

I can go into 'More settings' and scroll down all the way to the bottom where it says 'Reset settings', 'Export settings' etc, but nothing there or on the way down in 'Settings' is there anything to clearly indicate how to keep the sidebar away/keep VDH in the popup only.

I can click any combination of the tickboxes and none of them keep the sidebar away on open (including 'Use legacy UI').

I can do all this from both the sidebar 'More settings', and the popup 'More settings', never click the 'Show in sidebar' button again, and when I reopen Firefox, the sidebar is there.

I can uninstall VDH, restart everything, install VDH, and the sidebar is there again when I restart Firefox (without having pressed 'Show in sidebar' or anything else.

So, can you please tell me what I've missed in making sure this sidebar never shows up when I start Firefox? What buttons or tickboxes should I be pressing or not pressing?

**Reply by GrampaWildWilly** [reply]
The sidebar is there when VDH is disabled?  Now that's seriously weird.  You don't have VDH installed twice, do you?  I didn't think that was possible but . . .  You don't have 2 entries for VDH in about:addons, do you?  I suppose you're not doing anything wrong.  But you're not getting it to behave the same as I am.  It sounds to me like you're doing everything that should cause Firefox to pack VDH away in its normal drop-down menu . . . except that's not happening.  I haven't seen any specific settings, not checkboxes nor any other type, that deal with the sidebar.  The only thing I've ever seen is that in the drop-down menu, there's the button for Show in sidebar, & in the sidebar, there's the button for Show in popup.  Those are the only controls I know of that deal with the sidebar.  For me, swapping the sidebar back to the menu leaves VDH persistently in the menu, not the sidebar.

I think this is beyond my ability to diagnose.  I haven't actually said this here but I'm sure you read it in the other discussions.  I do not work for VDH.  I'm just another user.  I think we need to get @paulrouget Paul in here.  He's the developer.  I'm stumped.

**Reply by GrampaWildWilly** [reply]
This might work, it might not.  There is a button for `Export settings` at the bottom of the VDH `More settings` page.  I've just used it for the first time & discovered that even though the button looks like it's done nothing, it actually writes a json file containing the settings in the directory identified in the settings as the `Download directory`.  Used to be this function launched a file selection dialog.  Not any more.  Anyway, using your text editor open the settings.json file VDH exports.  It's just a plain text file so you can do that.  The file no longer contains your E-mail address & license key, another change from past releases, & a distinct improvement in my opinion.  That would make it safe to post without redacting it.  Not that I think you should post it.  Paul might ask you to but I'm not saying that.  Down near the bottom of my exported settings, I have this line:

`    "use_sidebar": false,`

What do you have?  If it says `true` for you, you might try this.  It's on your head if something screws up.  I haven't personally tried this so it's up to you if you want to do it.  I'm thinking your line says `true` instead of `false`.  So just change it to `false` & import the settings back in.  After that, just for good measure.  I would close all your Firefox windows & then launch Firefox again.  But like I say, I haven't imported my settings in this version of VDH so you'd be pioneering it.  IF you decide to do it.  IF.  I wouldn't blame you if you decide not to.  But at least export your settings & see what you have in there.

If your line looks the same as mine, then I'm even more stumped than I was.

**Reply by s3cret000** [reply]
"use_sidebar" was set at true. Changed it to false. Restarted. Sidebar still showing up. Exported settings, still set at false.

**Reply by GrampaWildWilly** [reply]
That is every kind of contradictory conditions possible.  There has to be a bug in there somewhere.  You're making me even more grateful I'm not seeing the problem.  Like that's any comfort to you.  I hope Paul shows up here soon.

**Comment by paulrouget** [comment]
Hello.

The sidebar bug is actually a Firefox bug, not a VDH bug :(

We are working with Mozilla on addressing this issue.

**Reply by GrampaWildWilly** [reply]
![#01](https://github.com/user-attachments/assets/4fe2e8b9-7580-44cf-af3c-5d16203fd7e3)

**Reply by premium9000** [reply]
omg I thought I was going crazy. the sidebar always opens right when I open Firefox. Then to further my dismay, the sidebar opens every time I open a new tab! How on earth do I disable it?

@GrampaWildWilly I close the sidebar by clicking that popup button, but I guess it resets and the sidebar opens again on startup and when opening new tabs. such an annoying issue.

**Reply by premium9000** [reply]
@paulrouget any idea on how long it might be until the sidebar issue gets resolved? I think in the meantime I might go back to using Google Chrome instead of Firefox.

**Reply by GrampaWildWilly** [reply]
Do you have either History or Bookmarks set to always open in a sidebar?  On my system, it is enough to close either of those Firefox built-in sidebars to stop them from appearing at startup or on open of a new tab.  I've got Firefox 129.0.2 64-bit.  And going back to using Chrome because of a sidebar issue seems a little extreme, but to each his own, I suppose.

**Comment by GrampaWildWilly** [comment]
I have just added a comment to #1192 that is relevant to this discussion.

**Comment by knallie** [comment]
The issue came back again with the recent Firefox versions 134.0 and some versions before.
Tested with VDH version 9.1.0.44.
Is there any known solution to fix the bug in the meantime available?

**Reply by knallie** [reply]
There is the solution: 
https://github.com/aclap-dev/video-downloadhelper/discussions/1192#discussioncomment-10770456

It worked for me fine.
